docs/whats_new/v0.4.0.md

```diff
@@ -0,0 +1,152 @@
+# Version 0.4.0
+
+v0.4.0 contains major enhancements to Tabular and Text modules, along with many quality of life improvements and fixes.
+
+TODO: Finalize high level description, add links to documentation, add Text module updates, add links to PRs
+
+This release contains **149** commits from **14** contributors!
+
+Special thanks to @zhiqiangdon, @willsmithorg, @DolanTheMFWizard, @truebluejason, @killerSwitch, and @Xilorole who were first time contributors to AutoGluon this release!
+
+Full Contributor List (ordered by # of commits):
+
+@Innixma, @yinweisu, @gradientsky, @zhiqiangdon, @jwmueller, @willsmithorg, @sxjscience, @DolanTheMFWizard, @truebluejason, @taesup-aws, @Xilorole, @mseeger, @killerSwitch, @rschmucker
+
+# Changes
+
+## General
+
+- AutoGluon now supports Windows OS! Both CPU and GPU are supported on Windows. @Innixma, @yinweisu (#1463, #1470)
+- AutoGluon now supports Python 3.9! @yinweisu (#1330)
+- Python 3.6 support has been dropped.
+- AutoGluon no longer depends on MXNet. PyTorch is now used for all deep learning models. (@zhiqiangdon, @sxjscience, @yinweisu, @jwmueller, @Innixma)
+- AutoGluon no longer depends on ConfigSpace, cython, dill, paramiko, autograd, openml, d8, graphviz and many more packages.
+This greatly simplifies installation of AutoGluon, particularly on Windows. @Innixma
+- Entirely refactored HPO logic to break dependencies on ConfigSpace and improve stability and ease of development.
+HPO has been simplified to use random search in this release while we work on
+re-introducing the more advanced HPO methods such as bayesopt in a future release.
+Additionally, removed 40,000 lines of out-dated code to streamline future development.
+@Innixma (#1397, #1411, #1414, #1431, #1443, #1511)
+- Added cloud training with AWS SageMaker tutorials. @gradientsky (#1375)
+- Added GPU training tutorial for tabular models. @gradientsky (#1527)
+- Added feature preprocessing tutorial for tabular. @willsmithorg (#1478)
+- Added `autogluon.common` to simplify dependency management for future submodules. @Innixma (#1386)
+- Removed `autogluon.mxnet` and `autogluon.extra` submodules as part of code cleanup. @Innixma (#1397, #1411, #1414)
+- Refactored logging to avoid interfering with other packages. @yinweisu (#1403)
+- Fixed logging output on Kaggle, previously no logs would be displayed while fitting AutoGluon in a Kaggle kernel. @Innixma (#1468)
+- Added platform tests for Linux, MacOS, and Windows. @yinweisu (#1464, #1506, #1513)
+- Added ROADMAP.md to highlight past, present, and future feature prioritization and progress to the community. @Innixma (#1420)
+- Various documentation and CI improvements
```

Comment:
I feel that the release note is too long. We may just list the PRs and not list the authors (because each PR will have a author).

Reply:
I think it is good practice to credit all authors in the release notes for their work. For example, this is done in scikit-learn: https://scikit-learn.org/stable/whats_new/v1.0.html#changes-1-0

Reply:
I think we may thank all the contributors. The current change-log is way too long... The key message of the release is not very clear.

Reply:
Another two examples are TVM release note: https://github.com/apache/tvm/releases/tag/v0.8.0, PyTorch release note: https://github.com/pytorch/pytorch/releases (See the long-list of PRs...) . For me, I just feel that the release note is not very focused.

Reply:
Updated, please see revised version